enderio-machines/src/main/java/com/enderio/machines/client/gui/widget/ChannelSelectWidget.java

```diff
@@ -0,0 +1,318 @@
+        this.sendChannelsWidget = createChannelListWidget(config.sendChannels(),
+            sideX, config.y() + 48, 104, 35);
+        this.receiveChannelsWidget = createChannelListWidget(config.receiveChannels(),
+            sideX, config.y() + 48 + 35 + 20, 104, 35);
+
+        this.channelNameBox = createChannelNameBox(config.x() + 25, config.y() + 12);
+        this.privateButton = createPrivateButton(config.x() + 118, config.y() + 12);
+        this.addChannelButton = createAddChannelButton(config.x() + 137, config.y() + 12);
+        this.deleteChannelButton = createDeleteChannelButton(config.x() + 91, config.y() + 142);
+        this.sendButton = createSendButton(config.x() + 119, config.y() + 58);
+        this.receiveButton = createReceiveButton(config.x() + 119, config.y() + 113);
+
+
+        Collections.addAll(children,
+            availableChannelsWidget, sendChannelsWidget, receiveChannelsWidget,
+            channelNameBox, privateButton, addChannelButton, deleteChannelButton,
+            sendButton, receiveButton
+        );
+    }
+
+
+    private ChannelListWidget createChannelListWidget(Set<Channel> channels, int x, int y, int width, int height) {
+        ChannelListWidget widget = new ChannelListWidget(Minecraft.getInstance(), width, height, 0, 17);
+        widget.setPosition(x, y);
+        widget.setChannels(channels);
+        return widget;
+    }
+
+    private EditBox createChannelNameBox(int x, int y) {
+        EditBox box = new EditBox(Minecraft.getInstance().font, x, y, 87, 18, Component.literal("ChannelName"));
+        box.setCanLoseFocus(true);
+        box.setTextColor(0xFFFFFFFF);
+        box.setTextColorUneditable(0xFFFFFFFF);
+        box.setBordered(true);
+        box.setMaxLength(CHANNEL_NAME_MAX_LENGTH);
+        box.setEditable(true);
+        return box;
+    }
+
+    private ToggleIconButton createPrivateButton(int x, int y) {
+        return new ToggleIconButton(x, y, 16, 16,
+            (b) -> b ? ICON_CHANNEL_PRIVATE : ICON_CHANNEL_PUBLIC,
+            (b) -> b ? EIOLang.CHANNEL_PRIVATE : EIOLang.CHANNEL_PUBLIC,
+            channelOps.isPrivate(),
+            channelOps::setPrivate
+        );
+    }
+
+    private IconButton createAddChannelButton(int x, int y) {
+        return new IconButton(x, y, 16, 16, ICON_CHANNEL_ADD, EIOLang.ADD_CHANNEL, this::addChannel);
+    }
+
+    private IconButton createDeleteChannelButton(int x, int y) {
+        return new IconButton(x, y, 16, 16, ICON_CHANNEL_DELETE, EIOLang.DELETE_CHANNEL, this::removeSelectedChannel);
+    }
+
+    private IconButton createSendButton(int x, int y) {
+        return new IconButton(x, y, 16, 16, ICON_ARROWS, Component.empty(), this::handleSendButton);
+    }
+
+    private IconButton createReceiveButton(int x, int y) {
+        return new IconButton(x, y, 16, 16, ICON_ARROWS, Component.empty(), this::handleReceiveButton);
+    }
+
+    private ChannelOperations createChannelOperations(TransceiverScreen screen) {
```

Comment:
is there a reason that ChannelOperations is an interface with a single anonymous impl? Can it be turned into a normal class?